rockcraft/plugins/python_plugin.py

```diff
@@ -17,13 +17,43 @@
 """The Rockcraft Python plugin."""
 
 import logging
-from typing import Optional
+from textwrap import dedent
+from typing import List, Optional
 
 from craft_parts.plugins import python_plugin
 from overrides import override
 
 logger = logging.getLogger(__name__)
 
+# Template for the sitecustomize module that we'll add to the payload so that
+# the pip-installed packages are found regardless of how the interpreter is
+# called.
+SITECUSTOMIZE_TEMPLATE = dedent(
+    """
+    # sitecustomize added by Rockcraft.
+    import site
+    import sys
+
+    major, minor = sys.version_info.major, sys.version_info.minor
+    site_dir = f"/lib/python{major}.{minor}/site-packages"
+    dist_dir = "/usr/lib/python3/dist-packages"
+
+    # Add the directory that contains the venv-installed packages.
+    site.addsitedir(site_dir)
+
+    # Make sure that this site-packages dir comes *before* the base-provided
+    # dist-packages dir in sys.path.
+    path = sys.path
+    site_index = path.index(site_dir)
+    dist_index = path.index(dist_dir)
```

Comment:
you probably want to catch a ValueError, I don't think self compiled python has a dist_dir.